ksql-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -273,11 +275,10 @@ private DataSourceNode buildNonJoinNode(final List<AliasedDataSource> sources) {
       final BiFunction<Expression, String, Boolean> matcher
   ) {
     for (int i = 0; i < analysis.getSelectExpressions().size(); i++) {
-      final Expression expression = analysis.getSelectExpressions().get(i);
-      final String alias = analysis.getSelectExpressionAlias().get(i);
+      final SelectExpression select = analysis.getSelectExpressions().get(i);
 
-      if (matcher.apply(expression, alias)) {
-        return Optional.of(alias);
+      if (matcher.apply(select.getExpression(), select.getName())) {
+        return Optional.of(select.getName());
```

Comment:
The `name` of `SelectExpression` cannot be null right? It's not defined as `Optional` in the class. Why does this return `Optional` here?

Reply:
It's looking for a projection of the field from the value that was annotated as the key, e.g.:
```
create stream foo (k string, v int) with (key='k', ...);
create stream bar as select k as nk from fool
```